internal/middleware/ratelimiter.go

```diff
@@ -78,28 +81,40 @@ func RateLimiter(client *db.DiceDB, next http.Handler, limit int64, window float
 			}
 		}
 
-		// Get the cron last cleanup run time
-		var lastCronCleanupTime int64
-		resp := client.Client.Get(ctx, utils.LastCronCleanupTimeUnixMs)
-		if resp.Err() != nil && !errors.Is(resp.Err(), dicedb.Nil) {
-			slog.Error("Failed to get last cron cleanup time for headers", slog.Any("err", resp.Err().Error()))
-		}
-
-		if resp.Val() != "" {
-			lastCronCleanupTime, err = strconv.ParseInt(resp.Val(), 10, 64)
-			if err != nil {
-				slog.Error("Error converting last cron cleanup time", "error", err)
-			}
+		secondsDifference, err := calculateNextCleanupTime(ctx, client, cronFrequencyInterval)
+		if err != nil {
```

Comment:
In case of `err` should we set header val to `-1` or something so that FE can not update the cleanup time in that case?

Reply:
Done @lucifercr07 !